### PR TITLE
Add filters

### PR DESCRIPTION
**Pull request by evesonr**
This PR:
- adds filters based on metadata such as primary organisation
- adds filters on page paths, including an option to filter on parent page paths
- tidies up use of Qdrant and HuggingFace environment variables
- adds additional metadata from BigQuery
- adds code to pull unique values of metadata from BigQuery to use in dropdown filters and parent page filtering
- adds script to download metadata values from BigQuery to populate dropdowns

### CODE REVIEW

File: collection/create_collection.py [hunk 1]
```diff
@@ -26,7 +29,7 @@
     query_read,
 )
 
-client = QdrantClient(os.getenv("QDRANT_HOST"), port=6333)
+client = QdrantClient(QDRANT_HOST, port=QDRANT_PORT)
```
Comment (hwrightson) [inline collection/create_collection.py:32]: 👀

File: src/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import json
+
+
+def jsonify_data(records: list, labelled=False):
+    """
+    Create json string from feedback
+    :return: json string of feedback records
+    """
+    subs = []
+    for i, item in enumerate(records):
```
Comment (hwrightson) [inline src/utils/utils.py:11]: Out of interest, why enumerate here given i isn't used anywhere?

File: src/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                "id": item["feedback_record_id"],
+                "feedback": response_value,
+                "label": [item["labels"] if labelled else ""],
+            }
+        )
+
+    return json.dumps(subs, indent=4)
+
+
+def process_txt_file(file_obj):
```
Comment (hwrightson) [inline src/utils/utils.py:24]: Very nice!

**Review by hwrightson — APPROVED** [review]
LGTM, couple of comments which are more questions but everything looks good. Some very clever functionality and dynamic programming used throughout!

File: app/main.py [hunk 4]
```diff
@@ -27,7 +34,7 @@
 # TODO: Replace with call to HF Inferece API or OpenAI API
 @st.cache_resource()
 def load_qdrant_client(port):
-    client = QdrantClient(os.getenv("QDRANT_HOST"), port=port)
+    client = QdrantClient(QDRANT_HOST, port=port)
```
[the remark this answers is not in the document]
Comment (evesonr) [reply app/main.py]: done!

File: src/utils/utils.py (hunk 2 above)
Comment (evesonr) [reply src/utils/utils.py]: removed :)

File: app/main.py [hunk 5]
```diff
@@ -38,40 +45,110 @@ def load_model(model_name):
...
+@st.cache_resource()
+def load_filter_dropdown_values(path_to_json):
+    with open(path_to_json, "r") as file:
+        data = json.load(file)
+    return data
+
 
-# Set any static variables for filtering retrieval
-filter_key = "subject_page_path"
+client = load_qdrant_client(QDRANT_PORT)
```
[the remark this answers is not in the document]
Comment (evesonr) [reply app/main.py]: sorted

**Pull request merged**